File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/primitive/LongFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.primitive;

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface LongFilter {

    AnnotationState status() default AnnotationState.ON;

    long lessThan() default Long.MAX_VALUE;

    long lessOrEqual() default Long.MAX_VALUE;

    long moreThan() default Long.MIN_VALUE;

    long moreOrEqual() default Long.MIN_VALUE;

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/primitive/LongFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.primitive

import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

/**
 * [LongFilter] annotation is used to filter incoming [Long] value
 *
 * @param lessThan - passes only less values
 *
 * @param lessOrEqual - passes  less and equal values
 *
 * @param greatThan - passes only great values
 *
 * @param greatOrEqual - passes great and equals values
 *
 */
@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class LongFilter(
        val status: AnnotationState = AnnotationState.ON,
        val lessThan: Long = java.lang.Long.MAX_VALUE,
        val lessOrEqual: Long = java.lang.Long.MAX_VALUE,
        val greatThan: Long = java.lang.Long.MIN_VALUE,
        val greatOrEqual: Long = java.lang.Long.MIN_VALUE
)
